src/timer.tsx

```diff
@@ -0,0 +1,438 @@
+'use client';
+import classNames from 'classnames';
+import * as React from 'react';
+import Box from './box';
+import {useAriaId, useIsomorphicLayoutEffect, useTheme} from './hooks';
+import Inline from './inline';
+import ScreenReaderOnly from './screen-reader-only';
+import {Text2, Text6} from './text';
+import {ThemeVariant, useThemeVariant} from './theme-variant-context';
+import * as styles from './timer.css';
+import {getPrefixedDataAttributes} from './utils/dom';
+import {isEqual} from './utils/helpers';
+import {isRunningAcceptanceTest} from './utils/platform';
+
+import type {DataAttributes} from './utils/types';
+
+const DAY_IN_HOURS = 24;
+const HOUR_IN_MINUTES = 60;
+const MINUTE_IN_SECONDS = 60;
+const SECOND_IN_MS = 1000;
+
+const MINUTE_IN_MS = MINUTE_IN_SECONDS * SECOND_IN_MS;
+const HOUR_IN_MS = HOUR_IN_MINUTES * MINUTE_IN_MS;
+const DAY_IN_MS = DAY_IN_HOURS * HOUR_IN_MS;
+
+export type TimeUnit = 'days' | 'hours' | 'minutes' | 'seconds';
+type Label = 'none' | 'short' | 'long';
+
+interface Timestamp {
```

Comment:
this is a bad name, a timestamp is the number of seconds since 1970, so it's weird to be represented with this data structure.

I'd rename this to something like `RemainingTime` or similar

Reply:
fixed